### PR TITLE
Freeze dependencies

### PR DESCRIPTION
**Pull request by jweede**
Freeze dependencies via `[versions]` for reproducible builds.

Also added things that appeared to be missing from `.gitignore`, and made the ignore statements more specific

### CODE REVIEW

File: buildout.cfg [hunk 1]
```diff
@@ -49,3 +49,86 @@ recipe = zc.recipe.egg
 eggs = keas.kmi
 
 [versions]
+[versions]
+setuptools = 32.2.0
```
Comment (vincentfretin) [inline buildout.cfg]: careful, you have [versions] twice

Comment (jweede) [reply buildout.cfg]: Whoops. Thanks for catching that!

**Review by mgedmin — COMMENTED** [review]
I'm -0 for this change -- if we freeze all the versions, how will we know that keas.kmi is no longer compatible with newest versions of its dependencies?  We can't expect every user of keas.kmi to use our versions.cfg -- this doesn't scale.

**Comment by vincentfretin** [comment]
Maybe we can just add a versions.cfg file but don't actually use it? Just to have a set of packages versions that we know is working.

**Comment by mgedmin** [comment]
A versions.cfg that's not used may get out of sync, and then what use is it?

IMHO it would be more useful to make sure Travis prints all the picked versions (e.g. like `tox` does) so one could look up the last successful build and see what was being used.

**Comment by vincentfretin** [comment]
Yeah ok, I like the idea of printing the versions used in the tests.

**Comment by jweede** [comment]
Hi @mgedmin, @vincentfretin. Thanks for taking a look at this.

The issue at hand in this PR is **reproducible builds**. I didn't file this request from idle speculation, but from a real-world problem we had with v3.1.1 of keas.kmi. As you know external dependencies update from time-to-time and often break old code. Thus I wanted to introduce versions.cfg so that when a release is cut, we know the set of versions that worked with it for all time.

I completely agree that we must continue to test new versions of external dependencies against this code, but it's essential that we have a build flow that is reproducible. 

I have a rough idea of how to balance reproducible builds with testing against latest deps using pip and setup.py, but I'm not sure what we're supposed to do with a buildout workflow. Is there a "best practices" page?

**Comment by mgedmin** [comment]
Please do not interpet my -0 as a veto -- it's merely my personal opinion.  If having a versions.cfg helps you, feel free to merge!

**Comment by icemac** [comment]
@jweede Maybe you should sign a Zope contributor agreement, so you can merge this PR yourself.

**Comment by jweede** [comment]
Closing PR. We ended up pinning dependencies with a docker container. Let me know if anyone's interested in merging our Dockerfile.

**Pull request closed without merging**